### ISSUE TITLE
Push to mirror repository fails with "fatal: --mirror can't be combined with refspecs"

### ISSUE THREAD

**Issue by leo-liar** [sourcegit-scm/sourcegit#844]
Hello,

I've set up a bare mirror repository on a network share for my local repo. Mirror pushing for this remote is enabled via `remote.origin.mirror=true` in my local repos `config` file.

When trying to push to this remote in SourceGit this immediatly fails with the error message:
```
fatal: --mirror can't be combined with refspecs
```

When trying to push via command line this works without any problems.

---

EDIT:
I'm on SourceGit 8.44 portable on Windows 10. The aforementioned remote repo has just been made to a mirror repo, though I already did successfully push to it (after turning mirror mode on) before. In my local repo I did just some rebasing (rewording, squashing and reordering commits) prior to trying to push via SourceGit.

**Comment by love-linger** [comment]
I've just pushed a new commit that ignores the `remote.{REMOTE}.mirror` setting while pushing specified branch.

**Comment by love-linger** [comment]
You can download the latest CI build from Github Action.

**Comment by leo-liar** [comment]
Thanks for the quick response. I'll test this when I'm back home (after New Years).

For my understanding: Why are you ignoring the `mirror` setting? My repo local (as well as the remote) has only one branch (`main`). The mirror remote is also the only remote.

From what I read about git's mirror settings this should make an exact replica of my local repo (and remove anything that's not in local from the remote repo).

When I pushed via command line I just entered `git push` let handle git everything else and I was assuming that SourceGit does the same (if mirroring is enabled).

**Comment by love-linger** [comment]
* `SourceGit` do not support to push multiple branches.
* The `Push` popup will always select the current branch to push, and the `-c remote.origin.mirror=false` option is just used to prevent errors
* If there's only one branch in your repo, there're no differences between `git push --mirror` and `git -c remote.origin.mirror=false push --tags origin master:master`

**Comment by love-linger** [comment]
I'm sorry. I've reverted this changes. And for repo that enables `remote.{REMOTE}.mirror` settings, you can use custom actions to do this job.

**Comment by leo-liar** [comment]
> - If there's only one branch in your repo, there're no differences between git push --mirror and git -c remote.origin.mirror=false push --tags origin master:master

I don't think this does the same as `git push --mirror` as per [the docs](https://git-scm.com/docs/git-push#Documentation/git-push.txt---mirror).

On the other hand mirror pushing seems to have its own caveats for my intended use case (a bare backup repo on a networked share), according to some Stack Overflow discussions.

Nonetheless I think it would be nice if SourceGit would recognize and honor (and ideally show) a configured `remote.{REMOTE}.mirror` setting.

**Comment by love-linger** [comment]
> I don't think this does the same as git push --mirror as per [the docs](https://git-scm.com/docs/git-push#Documentation/git-push.txt---mirror).

Yes, you are right. This is why I rolled back the above commit after carefully reading the git documentation.

> Maybe this behaviour could be changed if a mirroring remote is configured/selected?

As I said, you can use `Custom Actions` to do this job. For example:

1. Add a new custom action in repository's configure
![image](https://github.com/user-attachments/assets/4e0f194b-d666-4b26-b14b-9319fb397080)
2. Use the created action to push
![image](https://github.com/user-attachments/assets/405f4596-f4d1-4dd1-9c37-36606bc53e73)

**Comment by love-linger** [comment]
Since you can add a remote as `--mirror=fetch` or `--mirror=push`, it's a little complex to cover all the conditions.

**Comment by leo-liar** [comment]
Ah, the custom action seems nice. I wasn't aware of that and will give it a try. (This will probably solve my issue.)
